fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,252 @@
+
+    public LakeTableTieringManager() {
+        this(new DefaultTimer("delay lake tiering", 1_000, 20));
+    }
+
+    @VisibleForTesting
+    protected LakeTableTieringManager(Timer lakeTieringScheduleTimer) {
+        this.lakeTieringScheduleTimer = lakeTieringScheduleTimer;
+        this.pendingTieringTables = new ArrayDeque<>();
+        this.liveTieringTableIds = new HashSet<>();
+        this.tablePathById = new HashMap<>();
+        this.tieringIntervalByTableId = new HashMap<>();
+        this.expirationReaper = new ExpiredOperationReaper();
+        expirationReaper.start();
+    }
+
+    public void initWithLakeTables(List<TableInfo> lakeTables) {
+        for (TableInfo tableInfo : lakeTables) {
+            // try to schedule the tables to be tiered, we wait for a period of TIMEOUT
+            // to avoid the existing tiering services are still tiering it
+            addLakeTable(tableInfo, TIERING_SERVICE_TIMEOUT_MS);
+        }
+    }
+
+    public void addLakeTable(TableInfo tableInfo) {
+        addLakeTable(tableInfo, null);
+    }
+
+    private void addLakeTable(TableInfo tableInfo, @Nullable Long delayMs) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.put(tableInfo.getTableId(), tableInfo.getTablePath());
+                    tieringIntervalByTableId.put(
+                            tableInfo.getTableId(),
+                            tableInfo.getTableConfig().getDataLakeTieringInterval().toMillis());
+                    scheduleTableTiering(tableInfo.getTableId(), delayMs);
+                });
+    }
+
+    private void scheduleTableTiering(long tableId) {
+        scheduleTableTiering(tableId, null);
+    }
+
+    private void scheduleTableTiering(long tableId, @Nullable Long delayMs) {
+        Long tieringInterval = tieringIntervalByTableId.get(tableId);
+        if (tieringInterval == null) {
+            // the table has been dropped, return directly
+            return;
+        }
+        lakeTieringScheduleTimer.add(
+                new DelayedTiering(tableId, delayMs == null ? tieringInterval : delayMs));
+    }
+
+    public void removeLakeTable(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.remove(tableId);
+                    tieringIntervalByTableId.remove(tableId);
+                });
+    }
+
+    @Nullable
+    public Tuple2<Long, TablePath> requestTable() {
+        Long tableId = pendingTieringTables.poll();
+        // no any pending table, return directly
+        if (tableId == null) {
+            return null;
+        }
+        TablePath tablePath = tablePathById.get(tableId);
+        // the table has been dropped, request again
+        if (tablePath == null) {
+            return requestTable();
+        }
+        tieringLivenessProbe(tableId);
+        return new Tuple2<>(tableId, tablePath);
+    }
+
+    private void tieringLivenessProbe(long tableId) {
```

Comment:
This method is not easy to understand. IIUC, this method just registers a timeout timer for the tiering table. Then we can call it `registerTieringTimeout()`.